test/testdata/todo/generics_should_fail.rb

```diff
@@ -0,0 +1,19 @@
+# typed: true
+
+class A1
+  extend T::Generic
+  TM = type_member
+
+  def self.foo
+    # should fail: TM is used in a static context
+    T::Array[TM].new
+  end
+end
+
+class A2
+  extend T::Generic
+  TM = type_member
+
+  # should fail: TM is used in a static context
+  s = T::Array[TM].new
+end
```

Comment:
I would be really surprised if the example `A2` ever differed from the example for `A1`, because it would mean that `self.<static-init>` is special in some way.

Reply:
Yes, I agree. There was some discussion about needing to handle `<static-init>` in a special way, so it's nice to have this here just as a canary if that ends up being necessary.